arp.rflx

```diff
@@ -0,0 +1,39 @@
+with Ethernet;
+with IPv4;
+
+package ARP is
+
+   -- RFC 826
+
+   type Hardware_Address_Space is
+   (
+      Hrd_Ethernet =>   1,
+      Hrd_HW_Exp1  =>  36,
+      Hrd_HW_Exp2  => 256
+   ) with Size => 16;
+
+   type Opcode is (Op_Request => 1, Op_Reply => 2) with Size => 16;
+
+   type Address_Length is mod 2**8;
+
+   type IPv4 is
+      message
+         Hrd : Hardware_Address_Space
+            then Pro if Hrd = Hrd_Ethernet;
+         Pro : Ethernet::Ether_Type;
+         HLn : Address_Length
+            --  ISSUE: Componolit/RecordFlux#559
+            --  then PLn if HLn = Ethernet::Address'Size / 8;
+            then PLn if HLn = 6;
+         PLn : Address_Length
+            --  ISSUE: Componolit/RecordFlux#559
+            --  then Op if PLn = IPv4::Address'Size / 8;
+            then Op if PLn = 4;
+         Op  : Opcode;
+         SHA : Ethernet::Address;
+         SPA : IPv4::Address;
+         THA : Ethernet::Address;
+         TPA : IPv4::Address;
+      end message;
```

Comment:
IMHO we should give ourselves a policy here. I know that the naming of the original ARP RFC is crap, but this will also be the document that's being consulted when trying to understand the protocol during implementation. I'd opt for names that are as consistent with the original spec as possible (except where prevented by syntax requirements).

I'll add some documentation to the cryptic stuff, though.

Reply:
Fixed in ace51fd.

Reply:
Do we also want to keep the RFC casing? I see the reason to keep the names as they are in the spec. But using their casing could lead to a multitude of different casings. Also an RFC might use different casings for different entities which we cannot express anyway.

Reply:
I changed the casing in 8bed33c.